Sources/Player/PlaybackEngine/Internal/AVQueuePlayer/Legacy/AVQueuePlayerWrapperLegacy.swift

```diff
@@ -88,7 +88,9 @@ final class AVQueuePlayerWrapperLegacy: GenericMediaPlayer {
 			guard let codec else {
 				return false
 			}
-			return supportedCodecs.contains(codec) || (codec == .FLAC && !isOfflined)
+			// Because we need to support items downloaded with both the legacy and the new OfflineEngine
+			// We can't just add .FLAC to the 'supportedCodecs' array, and we need to depend on the mediaType
+			return supportedCodecs.contains(codec) || (codec == .FLAC && mediaType != MediaTypes.BTS)
```

Comment:
does it make sense to extract this functionality from here and from `AVQueuePlayerWrapper` and avoid this logic duplication?

Reply:
`AVQueuePlayerWrapperLegacy` will be removed in a few weeks if the cache fixes have worked 🤞 . 
A feature flag right now decides whether we use one or the other.